File: C++/Jasio w szponach hazardu.cpp
#include <string>
#include <iostream>
#include <unordered_map>

void validate_score(const std::unordered_map<std::string, std::string>& bets, const std::unordered_map<std::string, int>& goal_count) 
{
    int correct_predictions = 0;
    for (const auto& bet : bets) 
    {
        const std::string& match = bet.first;
        const std::string& prediction = bet.second;
        size_t separator = match.find(" - ");
        std::string home = match.substr(0, separator);
        std::string away = match.substr(separator + 3);
        int home_goals = goal_count.count(home) ? goal_count.at(home) : 0;
        int away_goals = goal_count.count(away) ? goal_count.at(away) : 0;
        std::string result;
        if (home_goals > away_goals) 
            result = "1";
        else if (home_goals < away_goals) 
            result = "2";
        else 
            result = "0";
        if (prediction.find(result) != std::string::npos)
            ++correct_predictions;
    }
    if (correct_predictions == bets.size())
        std::cout << "TAK\n";
    else
        std::cout << "NIE " << correct_predictions << '/' << bets.size() << '\n';
}

int main() 
{
    int n;
    std::cin >> n;
    for (int i = 0; i < n; ++i) 
    {
        int m;
        std::cin >> m;
        std::cin.ignore();
        std::unordered_map<std::string, std::string> bets;
        for (int j = 0; j < m; ++j) 
        {
            std::string line;
            getline(std::cin, line);
            size_t last_space = line.rfind(' ');
            std::string match = line.substr(0, last_space);
            std::string bet = line.substr(last_space + 1);
            bets[match] = bet;
        }
        int q;
        std::cin >> q;
        std::cin.ignore();
        std::unordered_map<std::string, int> goal_count;
        for (int j = 0; j < q; ++j) 
        {
            std::string team;
            getline(std::cin, team);
            ++goal_count[team];
        }
        validate_score(bets, goal_count);
    }
    return 0;
}

File: Python/Jasio w szponach hazardu.py
def valide_score(bets, goal_count):
    correct_predictions = 0
    for match, bet in bets.items():
        home, away = match.split(' - ')
        home_goals = goal_count.get(home, 0)
        away_goals = goal_count.get(away, 0)
        if home_goals > away_goals:
            result = '1'
        elif home_goals < away_goals:
            result = '2'
        else:
            result = '0'
        if result in bet:
            correct_predictions += 1
    print('TAK' if correct_predictions == len(bets) else f'NIE {correct_predictions}/{len(bets)}')

n = int(input())
for _ in range(n):
    m = int(input())
    bets = {}
    for i in range(m):
        match, bet = input().rsplit(' ', 1)
        bets[match] = bet
    q = int(input())
    goal_count = {}
    for _ in range(q):
        team = input()
        goal_count[team] = goal_count.get(team, 0) + 1
    valide_score(bets, goal_count)